src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -74,15 +76,42 @@ export class WorkspaceIdConsumerWrapper {
           this.formatWorkspaceIdParams(wrapperOptions.request, options)
         ),
       delete: wrapperOptions.client.delete,
-      find: (options: SavedObjectsFindOptions) => {
-        return wrapperOptions.client.find(
-          // Based on https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/ui_settings/create_or_upgrade_saved_config/get_upgradeable_config.ts#L49
-          // we need to make sure the find call for upgrade config should be able to find all the global configs as it was before.
-          // It is a workaround for 2.17, should be optimized in the upcoming 2.18 release.
+      find: async (options: SavedObjectsFindOptions) => {
+        // Based on https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/ui_settings/create_or_upgrade_saved_config/get_upgradeable_config.ts#L49
+        // we need to make sure the find call for upgrade config should be able to find all the global configs as it was before.
+        // It is a workaround for 2.17, should be optimized in the upcoming 2.18 release.
+        const finalOptions =
           this.isConfigType(options.type as string) && options.sortField === 'buildNum'
             ? options
-            : this.formatWorkspaceIdParams(wrapperOptions.request, options)
-        );
+            : this.formatWorkspaceIdParams(wrapperOptions.request, options);
+        if (finalOptions.workspaces?.length) {
+          const workspaceList = await this.workspaceClient.list(
+            {
+              request: wrapperOptions.request,
+            },
+            {
+              perPage: 9999,
+            }
+          );
```

Comment:
nit: if there only have one workspace in `finalOptions.workspaces` we can use `this.workspaceClient.get`, may or may not have performance improvement.

Reply:
Yeah, we should do that.